js/foundation.core.utils.js

```diff
@@ -19,9 +19,13 @@ function rtl() {
  * @default {String} '' - if no plugin name is provided, nothing is appended to the uid.
  * @returns {String} - unique id
  */
-function GetYoDigits(length, namespace){
-  length = length || 6;
-  return Math.round((Math.pow(36, length + 1) - Math.random() * Math.pow(36, length))).toString(36).slice(1) + (namespace ? `-${namespace}` : '');
+function GetYoDigits(length = 6, namespace){
+  let str = '';
+  const chars = '0123456789abcdefghijklmnopqrstuvwxyz';
+  for (let i = 0; i < length; i++) {
+    str += chars[Math.floor(Math.random() * chars.length)];
```

Comment:
```suggestion
    str += chars[Math.floor(Math.random() * 36)];
```

Reply:
Either we cache it (do it once, the value never changes) or use directly `36`.

Reply:
I wouldn't use directly `36` because it's not that obvious it's the chars length (might introduce issue if someone extends the chars).

What do you mean with "cache it"?
Are you talking about something as `const charsLength = chars.length` before the for loop?

Reply:
Doesn't seem to make much difference if you use `String().length` more than once:
https://jsperf.com/whats-the-price-of-string-length

Reply:
Yes, because some VMs cache it but we should do that before. Also older browsers behave different. Newer ones like the current v8 engine are optimized.

Reply:
> Are you talking about something as `const charsLength = chars.length` before the for loop?

Yes, exactly. Please do it like this.

Reply:
alright, I've changed & pushed it